include/seqan3/range/view/single_pass_input.hpp

```diff
@@ -0,0 +1,402 @@
+ * \tparam urng_t The underlying range type.
+ * \implements input_range_concept
+ * \ingroup view
+ */
+template <input_range_concept urng_t>
+class single_pass_input_view : public detail::view_base
+{
+private:
+
+    //!\brief Friend declaration for seqan3::detail::single_pass_input_iterator.
+    template <typename view_t>
+    friend class single_pass_input_iterator;
+
+    //!\brief The pure range type without any reference type.
+    using pure_range_t       = std::remove_reference_t<urng_t>;
+    //!\brief The iterator type for the underlying range.
+    using urng_iterator_type = iterator_t<pure_range_t>;
+    //!\brief The sentinel type for the underlying range.
+    using sentinel           = single_pass_input_sentinel<sentinel_t<pure_range_t>>;
+
+    //!\brief The internal state.
+    struct state
+    {
+        //!\brief The underlying range.
+        urng_t             urng;
+        //!\brief The cached iterator of the underlying range.
+        urng_iterator_type cached_urng_iter{};
+    };
+
+    //!\brief Shared pointer of the data model.
+    std::shared_ptr<state> state_ptr{};
+
+public:
+    /*!\name Member types
+     * \{
+     */
+    //!\brief Always returns immutable reference type, since single_pass_input cannot change the underlying values.
+    using reference         = std::add_const_t<reference_t<remove_cvref_t<urng_t>>>;
+    //!\brief Value type.
+    using value_type        = value_type_t<pure_range_t>;
+    //!\brief Iterator type.
+    using iterator          = single_pass_input_iterator<single_pass_input_view>;
+    //!\brief Const iterator type. Same as iterator.
+    using const_iterator    = iterator;
+    //\}
+
+    /*!\name Constructor, destructor, and assignment.
+     * \{
+     * \brief All standard functions are explicitly defaulted.
+     */
+    constexpr single_pass_input_view() = default;
+    constexpr single_pass_input_view(single_pass_input_view const &) = default;
+    constexpr single_pass_input_view(single_pass_input_view &&) = default;
+    constexpr single_pass_input_view & operator=(single_pass_input_view const &) = default;
+    constexpr single_pass_input_view & operator=(single_pass_input_view &&) = default;
+    ~single_pass_input_view() = default;
+
+    //!\brief Construction from the underlying view.
+    single_pass_input_view(urng_t && urng) :
+        state_ptr{new state{std::forward<urng_t>(urng), urng.begin()}}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the current begin of the underlying range.
+     *
+     * \details
+     *
+     * Subsequent calls to begin will result in different positions if the iterator was incremented
+     * between the calls.
+     */
+    iterator begin()
+    {
+        return iterator{*this};
+    }
+
+    //!\brief Const version of begin is deleted, since the underlying state must be mutable.
+    const_iterator begin() const = delete;
```

Comment:
You define the `const_iterator` type, but then delete all functions that could return a `const_iterator` – that's a little inconsistent ;-)

I would also define the const_iterator as void.